code/game/objects/items/storage/dispenser.dm

```diff
@@ -0,0 +1,126 @@
+			continue
+		affecting.heal_overall_damage(2, 2, TRUE, TRUE)
+
+///checks if a mob that moved close is elligible to get heal beamed.
+/obj/machinery/deployable/dispenser/proc/entered_tiles(datum/source, mob/living/carbon/human/entering)
+	SIGNAL_HANDLER
+	if(!ishuman(entering) || !(entering.species.species_flags & ROBOTIC_LIMBS)) // can only affect robots
+		return
+	if(entering in affecting_list)
+		return
+	if(!line_of_sight(src, entering))
+		return
+
+	RegisterSignal(entering, COMSIG_PARENT_QDELETING, .proc/on_affecting_qdel)
+	entering.playsound_local(get_turf(src), 'sound/machines/dispenser/dispenser_heal.ogg', 50)
+	affecting_list[entering] = beam(entering, "blood_light", maxdistance = 3)
+	RegisterSignal(affecting_list[entering], COMSIG_PARENT_QDELETING, .proc/on_beam_qdel)
+
+///cleans human from affecting_list when it gets qdeletted
+/obj/machinery/deployable/dispenser/proc/on_affecting_qdel(datum/source)
+	SIGNAL_HANDLER
+	affecting_list -= source
+
+///cleans human from affecting_list when the beam gets qdeletted
+/obj/machinery/deployable/dispenser/proc/on_beam_qdel(datum/source)
+	SIGNAL_HANDLER
+	var/datum/beam/beam = source
+	UnregisterSignal(beam.target, COMSIG_PARENT_QDELETING)
+	affecting_list -= beam.target
+
+/obj/machinery/deployable/dispenser/attack_hand(mob/living/user)
+	. = ..()
+	var/obj/item/storage/internal_bag = internal_item
+	internal_bag.attack_hand(user)
+
+/obj/machinery/deployable/dispenser/attackby(obj/item/I, mob/user, params)
+	if(internal_item.attackby(I, user, params))
+		return
+	return ..()
+
+/obj/machinery/deployable/dispenser/MouseDrop(obj/over_object)
+	if(!ishuman(usr))
+		return
+	var/mob/living/carbon/human/user = usr //this is us
+	if(over_object != user || !Adjacent(user))
+		return
+	if(active != TRUE)
+		return
+	active = FALSE
+	balloon_alert_to_viewers("Undeploying...")
+	for(var/turf/turfs AS in RANGE_TURFS(2, src))
+		UnregisterSignal(turfs, COMSIG_ATOM_ENTERED)
+	for(var/mob/living/carbon/human/affecting AS in affecting_list)
+		qdel(affecting_list[affecting])
+		UnregisterSignal(affecting, COMSIG_PARENT_QDELETING)
+	affecting_list = null
+	STOP_PROCESSING(SSobj, src)
+	flick("dispenser_undeploy", src)
+	playsound(src, 'sound/machines/dispenser/dispenser_undeploy.ogg', 50)
+	addtimer(CALLBACK(src, .proc/disassemble, user), 4 SECONDS)
+
+/obj/machinery/deployable/dispenser/disassemble(mob/user)
+	var/obj/item/storage/internal_bag = internal_item
+	for(var/mob/watching in internal_bag.content_watchers)
+		internal_bag.close(watching)
+	return ..()
+
+/obj/item/storage/backpack/dispenser
+	name = "TX-9000 provisions dispenser"
+	desc = "The TX-9000 also known as \"Dispenser\" is a machine capable of holding a big amount of items on it, while also healing nearby humans. Your allies will often ask you to lay down one of those."
+	icon = 'icons/obj/items/storage/storage_48.dmi'
+	icon_state = "dispenser"
+	flags_equip_slot = ITEM_SLOT_BACK
+	max_w_class = 6
+	max_storage_space = 63
```

Comment:
Why 63 ? A standard backpack is 24 so this should probably be something like 48 to not be too busted